### PR TITLE
board: mimxrt1160_evk: linkserver support

### PR DESCRIPTION
**Pull request by hakehuang**
add linkserver support for mimxrt1160_evk

as when I run `pyocd list -t`, there is no mimxrt1160_evk support

```
 mimxrt1010                NXP                      MIMXRT1011xxxxx                         builtin  
  mimxrt1015                NXP                      MIMXRT1015xxxxx                         builtin  
  mimxrt1020                NXP                      MIMXRT1021xxxxx                         builtin  
  mimxrt1024                NXP                      MIMXRT1024xxxxx                         builtin  
  mimxrt1050                NXP                      MIMXRT1052xxxxB_hyperflash              builtin  
  mimxrt1050_hyperflash     NXP                      MIMXRT1052xxxxB_hyperflash              builtin  
  mimxrt1050_quadspi        NXP                      MIMXRT1052xxxxB_quadspi                 builtin  
  mimxrt1060                NXP                      MIMXRT1062xxxxA                         builtin  
  mimxrt1064                NXP                      MIMXRT1064xxxxA                         builtin  
  mimxrt1170_cm4            NXP                      MIMXRT1176xxxxx_CM4                     builtin  
  mimxrt1170_cm7            NXP                      MIMXRT1176xxxxx_CM7                     builtin  
```

for pyocd need install a cmsis-dap pack to support

### CODE REVIEW

**Comment by dleach02** [comment]
@hakehuang you need to update the board docs like @yvanderv did in #67344

**Comment by hakehuang** [comment]
> @hakehuang you need to update the board docs like @yvanderv did in #67344

@dleach02 add the linkserver in doc. please review again

File: boards/nxp/mimxrt1160_evk/board.cmake [hunk 1]
```diff
@@ -5,13 +5,12 @@
 #
 
 if(CONFIG_SOC_MIMXRT1166_CM7 OR CONFIG_SECOND_CORE_MCUX)
-board_runner_args(pyocd "--target=mimxrt1160_cm7")
 board_runner_args(jlink "--device=MIMXRT1166xxx6_M7" "--reset-after-load")
+board_runner_args(linkserver  "--device=MIMXRT1166xxxxx:MIMXRT1160-EVK")
 elseif(CONFIG_SOC_MIMXRT1166_CM4)
-board_runner_args(pyocd "--target=mimxrt1160_cm4")
 # Note: Please use JLINK above V7.50 (Only support run cm4 image when debugging due to default boot core on board is cm7 core)
```
Comment (mmahadevan108) [inline boards/nxp/mimxrt1160_evk/board.cmake:11]: How about for the CM4 core?

Comment (hakehuang) [reply boards/nxp/mimxrt1160_evk/board.cmake:11]: it is not working now for linkserver, let me forward you a mail

**Comment by decsny** [comment]
why removing pyocd?

**Comment by hakehuang** [comment]
> as when I run `pyocd list -t`, there is no mimxrt1160_evk support
> 
> ```
>  mimxrt1010                NXP                      MIMXRT1011xxxxx                         builtin  
>   mimxrt1015                NXP                      MIMXRT1015xxxxx                         builtin  
>   mimxrt1020                NXP                      MIMXRT1021xxxxx                         builtin  
>   mimxrt1024                NXP                      MIMXRT1024xxxxx                         builtin  
>   mimxrt1050                NXP                      MIMXRT1052xxxxB_hyperflash              builtin  
>   mimxrt1050_hyperflash     NXP                      MIMXRT1052xxxxB_hyperflash              builtin  
>   mimxrt1050_quadspi        NXP                      MIMXRT1052xxxxB_quadspi                 builtin  
>   mimxrt1060                NXP                      MIMXRT1062xxxxA                         builtin  
>   mimxrt1064                NXP                      MIMXRT1064xxxxA                         builtin  
>   mimxrt1170_cm4            NXP                      MIMXRT1176xxxxx_CM4                     builtin  
>   mimxrt1170_cm7            NXP                      MIMXRT1176xxxxx_CM7                     builtin  
> ```

@decsny 

see my comments in this pr

```
as when I run pyocd list -t, there is no mimxrt1160_evk support

 mimxrt1010                NXP                      MIMXRT1011xxxxx                         builtin  
  mimxrt1015                NXP                      MIMXRT1015xxxxx                         builtin  
  mimxrt1020                NXP                      MIMXRT1021xxxxx                         builtin  
  mimxrt1024                NXP                      MIMXRT1024xxxxx                         builtin  
  mimxrt1050                NXP                      MIMXRT1052xxxxB_hyperflash              builtin  
  mimxrt1050_hyperflash     NXP                      MIMXRT1052xxxxB_hyperflash              builtin  
  mimxrt1050_quadspi        NXP                      MIMXRT1052xxxxB_quadspi                 builtin  
  mimxrt1060                NXP                      MIMXRT1062xxxxA                         builtin  
  mimxrt1064                NXP                      MIMXRT1064xxxxA                         builtin  
  mimxrt1170_cm4            NXP                      MIMXRT1176xxxxx_CM4                     builtin  
  mimxrt1170_cm7            NXP                      MIMXRT1176xxxxx_CM7                     builtin  
```

**Review by dleach02 — DISMISSED** [review]

**Comment by danieldegrasse** [comment]
@hakehuang I believe the RT1160 is supported via a CMSIS pack: https://pyocd.io/docs/target_support.html#cmsis-packs

```
$ pyocd pack find rt116
  Part              Vendor   Pack                 Version   Installed
-----------------------------------------------------------------------
  MIMXRT1165CVM5A   NXP      NXP.MIMXRT1165_DFP   17.0.0    False
  MIMXRT1165DVM6A   NXP      NXP.MIMXRT1165_DFP   17.0.0    False
  MIMXRT1165XVM5A   NXP      NXP.MIMXRT1165_DFP   17.0.0    False
  MIMXRT1166CVM5A   NXP      NXP.MIMXRT1166_DFP   17.0.0    False
  MIMXRT1166DVM6A   NXP      NXP.MIMXRT1166_DFP   17.0.0    False
  MIMXRT1166XVM5A   NXP      NXP.MIMXRT1166_DFP   17.0.0    False
```

**Comment by hakehuang** [comment]
> pyocd pack find rt116
>   Part              Vendor   Pack                 Version   Installed
> -----------------------------------------------------------------------
>   MIMXRT1165CVM5A   NXP      NXP.MIMXRT1165_DFP   17.0.0    False
>   MIMXRT1165DVM6A   NXP      NXP.MIMXRT1165_DFP   17.0.0    False
>   MIMXRT1165XVM5A   NXP      NXP.MIMXRT1165_DFP   17.0.0    False
>   MIMXRT1166CVM5A   NXP      NXP.MIMXRT1166_DFP   17.0.0    False
>   MIMXRT1166DVM6A   NXP      NXP.MIMXRT1166_DFP   17.0.0    False
>   MIMXRT1166XVM5A   NXP      NXP.MIMXRT1166_DFP   17.0.0    False

we need install this?
https://github.com/pyocd/cmsis-pack-manager
which is a big deal

**Comment by danieldegrasse** [comment]
> > ## pyocd pack find rt116
> > Part              Vendor   Pack                 Version   Installed
> > MIMXRT1165CVM5A   NXP      NXP.MIMXRT1165_DFP   17.0.0    False
> > MIMXRT1165DVM6A   NXP      NXP.MIMXRT1165_DFP   17.0.0    False
> > MIMXRT1165XVM5A   NXP      NXP.MIMXRT1165_DFP   17.0.0    False
> > MIMXRT1166CVM5A   NXP      NXP.MIMXRT1166_DFP   17.0.0    False
> > MIMXRT1166DVM6A   NXP      NXP.MIMXRT1166_DFP   17.0.0    False
> > MIMXRT1166XVM5A   NXP      NXP.MIMXRT1166_DFP   17.0.0    False
> 
> we need install this? https://github.com/pyocd/cmsis-pack-manager which is a big deal beofre doing that I meet
> 
> ```
>  pyocd pack find rt1160
> 0000281 I No pack index present, downloading now... [pack_cmd]
> 0001784 W No matching devices. Please make sure the pack index is up to date. [pack_cmd]
> ```

I do believe we need to install something, however I think a lot of our pyocd enablement already relies on packs. @butok could you clarify here? Should we keep pyocd enablement for a board if a pack needs to be downloaded to support flashing it?

**Comment by hakehuang** [comment]
> I do believe we need to install something, however I think a lot of our pyocd enablement already relies on packs. @butok could you clarify here? Should we keep pyocd enablement for a board if a pack needs to be downloaded to support flashing it?

looks like the pack update is can not work behide the firewall, which makes things difficult, let me keep the pyocd options but still need clarify how to enable it.

**Comment by butok** [comment]
> > I do believe we need to install something, however I think a lot of our pyocd enablement already relies on packs. @butok could you clarify here? Should we keep pyocd enablement for a board if a pack needs to be downloaded to support flashing it?
> 
> looks like the pack update is can not work behide the firewall, which makes things difficult, let me keep the pyocd options but still need clarify how to enable it.

PYOCD support should be provided for any platform that has the correct CMSIS DFP package.

Installing the CMSIS DFP package is standard practice for pyocd and is described in its documentation.
If automatic installation is blocked by a firewall, the package can be downloaded from https://www.keil.arm.com/packs/.

**Comment by hakehuang** [comment]
@danieldegrasse  @butok  @decsny , I add  back the pyocd support back, can you help to review again. Thanks in advance.

**Comment by decsny** [comment]
do we intend to be missing a documentation blurb about pyocd?

**Comment by dleach02** [comment]
> do we intend to be missing a documentation blurb about pyocd?

Lets handle that with a separate PR since it is generic to pyocd. This PR is just adding linkserver and got a bit wrapped up with removing then returning the pyocd.

**Review by dleach02 — APPROVED** [review]

**Comment by danieldegrasse** [comment]
> A debug probe is used for both flashing and debugging the board. This board is configured by default to use the [OpenSDA DAPLink Onboard Debug Probe](https://builds.zephyrproject.io/zephyr/pr/72505/docs/develop/flash_debug/probes.html#opensda-daplink-onboard-debug-probe), however the [pyOCD Debug Host Tools](https://builds.zephyrproject.io/zephyr/pr/72505/docs/develop/flash_debug/host-tools.html#pyocd-debug-host-tools) do not yet support programming the external flashes on this board so you must reconfigure the board for one of the following debug probes instead.

Do we want to update this blub? Or do we still need to update the onboard debug firmware from the default in order to use LinkServer?

**Comment by decsny** [comment]
> > do we intend to be missing a documentation blurb about pyocd?
> 
> Lets handle that with a separate PR since it is generic to pyocd. This PR is just adding linkserver and got a bit wrapped up with removing then returning the pyocd.

no, what I mean is, this PR is adding pyocd to the board, we usually put documentation about all the debug  probes supported for the board in the board doc page, except the reason I'm asking is I don't know if pyocd is an exception for this

**Review by decsny — APPROVED** [review]

**Comment by hakehuang** [comment]
> > A debug probe is used for both flashing and debugging the board. This board is configured by default to use the [OpenSDA DAPLink Onboard Debug Probe](https://builds.zephyrproject.io/zephyr/pr/72505/docs/develop/flash_debug/probes.html#opensda-daplink-onboard-debug-probe), however the [pyOCD Debug Host Tools](https://builds.zephyrproject.io/zephyr/pr/72505/docs/develop/flash_debug/host-tools.html#pyocd-debug-host-tools) do not yet support programming the external flashes on this board so you must reconfigure the board for one of the following debug probes instead.
> 
> Do we want to update this blub? Or do we still need to update the onboard debug firmware from the default in order to use LinkServer?

let me do this

**Pull request merged**